pythonforandroid/build.py

```diff
@@ -375,60 +379,61 @@ def prepare_build_environment(self,
                     ' a python 3 target (which is the default)'
                     ' then THINGS WILL BREAK.')
 
-        # This would need to be changed if supporting multiarch APKs
-        arch = self.archs[0]
-        toolchain_prefix = arch.toolchain_prefix
-        self.ndk_platform, ndk_platform_dir_exists = get_ndk_platform_dir(
-            self.ndk_dir, self.ndk_api, arch)
-        ok = ok and ndk_platform_dir_exists
-
         py_platform = sys.platform
         if py_platform in ['linux2', 'linux3']:
             py_platform = 'linux'
-        toolchain_versions, toolchain_path_exists = get_toolchain_versions(
-            self.ndk_dir, arch)
-        ok = ok and toolchain_path_exists
-        toolchain_versions.sort()
-
-        toolchain_versions_gcc = []
-        for toolchain_version in toolchain_versions:
-            if toolchain_version[0].isdigit():
-                # GCC toolchains begin with a number
-                toolchain_versions_gcc.append(toolchain_version)
-
-        if toolchain_versions:
-            info('Found the following toolchain versions: {}'.format(
-                toolchain_versions))
-            info('Picking the latest gcc toolchain, here {}'.format(
-                toolchain_versions_gcc[-1]))
-            toolchain_version = toolchain_versions_gcc[-1]
-        else:
-            warning('Could not find any toolchain for {}!'.format(
-                toolchain_prefix))
-            ok = False
-
-        self.toolchain_prefix = toolchain_prefix
-        self.toolchain_version = toolchain_version
-        # Modify the path so that sh finds modules appropriately
-        environ['PATH'] = (
-            '{ndk_dir}/toolchains/{toolchain_prefix}-{toolchain_version}/'
-            'prebuilt/{py_platform}-x86/bin/:{ndk_dir}/toolchains/'
-            '{toolchain_prefix}-{toolchain_version}/prebuilt/'
-            '{py_platform}-x86_64/bin/:{ndk_dir}:{sdk_dir}/'
-            'tools:{path}').format(
-                sdk_dir=self.sdk_dir, ndk_dir=self.ndk_dir,
-                toolchain_prefix=toolchain_prefix,
-                toolchain_version=toolchain_version,
-                py_platform=py_platform, path=environ.get('PATH'))
-
-        for executable in ("pkg-config", "autoconf", "automake", "libtoolize",
-                           "tar", "bzip2", "unzip", "make", "gcc", "g++"):
-            if not sh.which(executable):
-                warning(f"Missing executable: {executable} is not installed")
-
-        if not ok:
-            raise BuildInterruptingException(
-                'python-for-android cannot continue due to the missing executables above')
+
+        self.ndk_standalone = get_ndk_standalone(self.ndk_dir)
+        self.ndk_sysroot, ndk_sysroot_exists = get_ndk_sysroot(self.ndk_dir)
+        ok = ok and ndk_sysroot_exists
+        self.ndk_include_dir = join(self.ndk_sysroot, 'usr', 'include')
+
+        for arch in self.archs:
```

Comment:
Minor suggestion: this block below could have been move to its function to give more semantic and ease testing.
Then simply calling the function in the loop

Reply:
Addressed, but I have in mind a partial rewrite, will look at that a little bit deeper in the future.